server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -72,6 +73,8 @@
             (value) -> Translog.Durability.valueOf(value.toUpperCase(Locale.ROOT)), Property.Dynamic, Property.IndexScope);
     public static final Setting<Boolean> INDEX_WARMER_ENABLED_SETTING =
         Setting.boolSetting("index.warmer.enabled", true, Property.Dynamic, Property.IndexScope);
+    public static final Setting<Integer> INDEX_MAX_DOC_ID_LENGTH_SETTING =
+        Setting.intSetting("index.max_doc_id_length", 512, 1, Property.Dynamic, Property.IndexScope);
```

Comment:
The default value of 1 is a bit worrying. First of all, it gives very few docs possible. But also, it is smaller than our auto-id size. That would make for strange messages if using auto-generated ids.

I think I would prefer to just leave the minimum at 512. Or at least at the max-size of an auto-generated id.